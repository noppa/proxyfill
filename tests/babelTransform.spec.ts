import traverse from './traverse'

describe('proxyfill Babel plugin', () => {
	it('should transform getters', () => {
		expect(traverse('foo.bar')).toMatchSnapshot()
	})
	it('should transform computed getters', () => {
		expect(
			traverse(`
			foo[0]
			foo[bar]
			foo[bar.baz]
		`)
		).toMatchSnapshot()
	})
	it('should transform setters', () => {
		expect(traverse('foo.bar = 42')).toMatchSnapshot()
	})
	it('should transform computed setters', () => {
		expect(
			traverse(`
			foo[0] = 42
			foo[bar] = 42
			foo[bar.baz] = 42
		`)
		).toMatchSnapshot()
	})
	it('should transform method calls', () => {
		expect(traverse('foo.bar()')).toMatchSnapshot()
	})
	it('should not transform direct function calls', () => {
		expect(traverse('bar()')).toMatchSnapshot()
	})
	it('should transform in operator usage', () => {
		expect(
			traverse(`
			foo in bar
			'foo' in bar
		`)
		).toMatchSnapshot()
	})
	it('should transform delete operator usage', () => {
		expect(traverse(`delete foo.bar`)).toMatchSnapshot()
	})
	it('should transform update member expressions', () => {
		expect(traverse('foo.bar++')).toMatchSnapshot()
	})
})
