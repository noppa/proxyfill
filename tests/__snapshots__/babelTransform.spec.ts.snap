// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proxyfill Babel plugin should not transform direct function calls 1`] = `
"import {
	Proxy,
	get as _proxyfillRuntime$get,
	set as _proxyfillRuntime$set,
	invoke as _proxyfillRuntime$invoke,
} from 'proxyfill/runtime'
bar()
"
`;

exports[`proxyfill Babel plugin should transform computed getters 1`] = `
"import {
	Proxy,
	get as _proxyfillRuntime$get,
	set as _proxyfillRuntime$set,
	invoke as _proxyfillRuntime$invoke,
} from 'proxyfill/runtime'

_proxyfillRuntime$get(foo, 0)

_proxyfillRuntime$get(foo, bar)

_proxyfillRuntime$get(foo, _proxyfillRuntime$get(bar, 'baz'))
"
`;

exports[`proxyfill Babel plugin should transform computed setters 1`] = `
"import {
	Proxy,
	get as _proxyfillRuntime$get,
	set as _proxyfillRuntime$set,
	invoke as _proxyfillRuntime$invoke,
} from 'proxyfill/runtime'

_proxyfillRuntime$set(foo, 0, 42)

_proxyfillRuntime$set(foo, bar, 42)

_proxyfillRuntime$set(foo, _proxyfillRuntime$get(bar, 'baz'), 42)
"
`;

exports[`proxyfill Babel plugin should transform getters 1`] = `
"import {
	Proxy,
	get as _proxyfillRuntime$get,
	set as _proxyfillRuntime$set,
	invoke as _proxyfillRuntime$invoke,
} from 'proxyfill/runtime'

_proxyfillRuntime$get(foo, 'bar')
"
`;

exports[`proxyfill Babel plugin should transform method calls 1`] = `
"import {
	Proxy,
	get as _proxyfillRuntime$get,
	set as _proxyfillRuntime$set,
	invoke as _proxyfillRuntime$invoke,
} from 'proxyfill/runtime'

_proxyfillRuntime$invoke(foo, 'bar', [])
"
`;

exports[`proxyfill Babel plugin should transform setters 1`] = `
"import {
	Proxy,
	get as _proxyfillRuntime$get,
	set as _proxyfillRuntime$set,
	invoke as _proxyfillRuntime$invoke,
} from 'proxyfill/runtime'

_proxyfillRuntime$set(foo, 'bar', 42)
"
`;
